core-bundle/src/Resources/contao/library/Contao/Search.php

```diff
@@ -159,6 +159,10 @@ public static function indexPage($arrData)
 		{
 			$arrData['keywords'] = trim(preg_replace('/ +/', ' ', StringUtil::decodeEntities($tags[1])));
 		}
+		else
+		{
+			$arrData['keywords'] = '';
+		}
```

Comment:
We should probably initialize this around line 61, shouldn't we?

Reply:
But lines 54 to 61 initialize `arrSet`. Setting `$arrData['keywords']` there would feel out-of-place to me.

I would alternatively suggest fixing it by `?? null` in line 174. Right now, this is the only place, where `$arrData['keywords']` is read.

Reply:
Good point. How about this:

```php
$arrData['keywords'] = '';

// Get the keywords
if (preg_match('/<meta[^>]+name="keywords"[^>]+content="([^"]*)"[^>]*>/i', $strHead, $tags))
{
	$arrData['keywords'] .= trim(preg_replace('/ +/', ' ', StringUtil::decodeEntities($tags[1])));
}

// Read the title and alt attributes
if (preg_match_all('/<* (title|alt)="([^"]*)"[^>]*>/i', $strBody, $tags))
{
	$arrData['keywords'] .= ' ' . implode(', ', array_unique($tags[2]));
}
```

Reply:
Done as suggested.